src/initialization.js

```diff
@@ -1,40 +1,22 @@
-        /* `settings` contains your SDK specific settings such as apiKey that your customer needs in order to initialize your SDK properly */
-
-        if (!testMode) {
-            /* Load your Web SDK here using a variant of your snippet from your readme that your customers would generally put into their <head> tags
-               Generally, our integrations create script tags and append them to the <head>. Please follow the following format as a guide:
-            */
+        if (!!window.branch) {
+            return;
+        }
 
-            // var clientScript = document.createElement('script');
-            // clientScript.type = 'text/javascript';
-            // clientScript.async = true;
-            // clientScript.src = 'https://www.clientscript.com/static/clientSDK.js';   // <---- Update this to be your script
-            // (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(clientScript);
-            // clientScript.onload = function() {
-            //     if (clientSDKObject && eventQueue.length > 0) {
-            //         // Process any events that may have been queued up while forwarder was being initialized.
-            //         for (var i = 0; i < eventQueue.length; i++) {
-            //             processEvent(eventQueue[i]);
-            //         }
-            //          // now that each queued event is processed, we empty the eventQueue
-            //         eventQueue = [];
-            //     }
-            //    clientSDKObject.initialize(forwarderSettings.apiKey);
-            // };
-        } else {
-            // For testing, you should fill out this section in order to ensure any required initialization calls are made,
-            // clientSDKObject.initialize(forwarderSettings.apiKey)
+        var branchScript = document.createElement('script');
+        branchScript.type = 'text/javascript';
+        branchScript.async = true;
+        branchScript.src = 'https://cdn.branch.io/branch-latest.min.js';
+        (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(branchScript);
+        branchScript.onload = function(){
+            (function(b,r,a,n,c,h,_,s,d,k){if(!b[n]||!b[n]._q){for(;s<_.length;)c(h,_[s++]);d=r.createElement(a);d.async=1;d.src="https://cdn.branch.io/branch-latest.min.js";k=r.getElementsByTagName(a)[0];k.parentNode.insertBefore(d,k);b[n]=h}})(window,document,"script","branch",function(b,r){b[r]=function(){b._q.push([r,arguments])}},{_q:[],_v:1},"addListener applyCode autoAppIndex banner closeBanner closeJourney creditHistory credits data deepview deepviewCta first getCode init link logout redeem referrals removeListener sendSMS setBranchViewData setIdentity track validateCode trackCommerceEvent logEvent disableTracking getBrowserFingerprintId crossPlatformIds lastAttributedTouchData".split(" "), 0);
```

Comment:
This snippet should be on the outside of the `onLoad` function. I don't think this resolves the async issue .

What you should do is move this code black back up above `branchScript.onload`, and then use the code that was part of the template that related to event queueing.    IE:

```
branchScript.onload = function() {
    branch.init(key, function(err, data) {
        isInitialized = true;
        if (eventQueue.length > 0) {
            for (var i = 0; i < eventQueue.length; i++) {
                processEvent(eventQueue[i]);
            }
        }
    });
}
}
```
so the above says, once branchScript loads, let's initialize, once branch initializes, go through the eventQueue.
I think this solution may resolve it. Basically, if branch isn't loaded yet, we queue the events in the `eventQueue`, then when it completely loads,

Reply:
I think what's happening with the async issue is that we're sometimes trying to do all of setup and tag creation stuff that happens in that line before the script loads properly. When that happens Branch never initializes properly so waiting to process events in the EventQueue until it does won't help.

In my testing the current setup does fix the async loading issue. I can't say I fully understand _why_ it works but it does seem to. :)

Reply:
i understand what's happening.  part of your snippet is being duplicated with functionality that we propose in our template. you'll notice that we are adding `https://cdn.branch.io/branch-latest.min.js` programmatically between lines 8-12.  but then your snippet basically does the same thing.  the two are probably colliding in a weird way, trying to load `branch-latest-min.js` twice which could result in the error. 

you can remove lines 9-14 here. then test that and you should be good to go. https://github.com/BranchMetrics/mparticle-javascript-integration-branch/blob/master/src/initialization.js

note that you `build/` folder is currently empty. running `KIT=Branch npm run watch` will create the file `build/Branch-Kit.js`. If you are okay with that name let me know and I can add that to our database. However, we need that file there before we can add it.

Reply:
Ah, good catch...seems super obvious in retrospect. I've made both fixes and merged into master, I think we should be good to go here.